toolchains/cheri_llvm/archive/README.md

```diff
@@ -0,0 +1,23 @@
+### Instructions for building the CHERI-LLVM toolchain archive for purecap riscv32
+
+This will use [cheribuild](https://github.com/CTSRD-CHERI/cheribuild) to build
+the CHERI-LLVM RISC-V 32bit "purecap" toolchain for embedded development.
+By default, the sources and toolchain will be fetched and built in `~/cheri/`.
+This can be changed using the `--source-root`, `--output-root`, `--build-root`,
+and `--tools-root` arguments. `-d` ensures required steps are executed (i.e.
+builds LLVM & Clang before compiling the libraries). `-j` controls the number
+of threads to be used for compilation.
+
+Clone the cheribuild repository anywhere (its directory will not be used for building), then:
+
+`./cheribuild.py newlib-baremetal-riscv32-purecap -d` (for libc, libm, libg)  
+`./cheribuild.py compiler-rt-builtins-baremetal-riscv32-purecap -d` (for builtins)
+
+Then `cd` to wherever the sdk folder is (`~/cheri/output/` by default,
+or within either of the directories pointed to when passing
+`--tool-root` or `--output-root`) and tar the `sdk` folder:  
+`tar -cvzf cheri_llvm_sdk.tar.gz sdk/`
+
+Lastly, copy `cheri_llvm_sdk.tar.gz` to this folder. The name must match the
+filename in the `archive` field in `../repository.bzl`
```

Comment:
We should discuss how to allow you to provide the compiler archive without requiring it to be in this repository or requiring the user to take a repository modifying action (e.g. copying the file into this subdir).  Modifying the repository as a build step is ["bad"](https://www.youtube.com/watch?v=wyKQe_i9yyo&t=54s).

Reply:
I've been thinking about how to allow you to supply the compiler archive without needing to ask the user to modify the repository.  There are two mechanisms I can think of:

1. Supply the repository yourself and forego the call to `cheri_llvm_repos()` in `crt_register_toolchains`.  In your WORKSPACE, you could then do something like this:
```
load("//third_party/crt:repos.bzl", "crt_repos")
crt_repos()
load("@crt//:repos.bzl", "crt_repos")
crt_repos()
load("@crt//:deps.bzl", "crt_deps")
crt_deps()

load("@crt//config:repo.bzl", "compiler_repository")
compiler_repository(
    name = "cheri_llvm_files",
    archive = "/path/to/your/cheri_llvm_sdk.tar.gz",
    exports = [ ... ],
)

load("@crt//config:registration.bzl", "crt_register_toolchains")
crt_register_toolchains(riscv32_cheri = True)
```

2. Modify the interface to `crt_register_toolchains` to allow you to pass the path to the compiler archive.  Your WORKSPACE would then have something like:
```
crt_register_toolchains(
    riscv32_cheri = "/path/to/your/cheri_llvm_sdk.tar.gz",
)
```
`crt_register_toolchains` can then pass the archive location on to `cheri_llvm_repos()` which will instantiate the `compiler_repository`.

I've been considering something like (2) for a while to enable end users to pass in alternate compilers, but I haven't yet had a need for it and I'd probably want to plumb things up a bit more to make the archive properties more configurable.